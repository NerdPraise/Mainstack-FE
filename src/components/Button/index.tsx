import { cn } from '@/lib/utils'

interface ButtonProps {
  children: React.ReactNode
  variant: 'dark' | 'light'
  onClick?: VoidFunction
  styles?: string
  disabled?: boolean
}

const variantClass = {
  dark: 'bg-black text-white hover:bg-[#373737]',
  light: 'bg-[#EFF1F6] text-black',
}

const Button = ({
  children,
  variant,
  disabled,
  onClick,
  styles,
}: ButtonProps) => {
  return (
    <button
      onClick={onClick}
      disabled={disabled}
      className={cn(
        'bg-black cursor-pointer duration-700 text-white py-4 rounded-full h-[52px] w-[168px]',
        variantClass[variant],
        styles,
        disabled &&
          'bg-[#DBDEE5] hover:bg-inherit cursor-not-allowed text-white'
      )}
    >
      {children}
    </button>
  )
}
export default Button
