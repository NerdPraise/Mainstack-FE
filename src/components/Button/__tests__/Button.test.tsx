import { describe, it, expect, vi } from 'vitest'
import { render, screen, fireEvent } from '@testing-library/react'
import Button from '../index'

describe('Button', () => {
  it('renders with default props', () => {
    render(<Button variant="dark">Click me</Button>)
    const button = screen.getByRole('button')
    expect(button).toBeInTheDocument()
    expect(button).toHaveTextContent('Click me')
    expect(button).toHaveClass('bg-black')
    expect(button).toHaveClass('text-white')
    expect(screen.debug).toMatchSnapshot()
  })

  it('renders with light variant', () => {
    render(<Button variant="light">Light Button</Button>)
    const button = screen.getByRole('button')
    expect(button).toHaveClass('bg-[#EFF1F6]')
    expect(button).toHaveClass('text-black')
  })

  it('handles click events', () => {
    const handleClick = vi.fn()
    render(
      <Button variant="dark" onClick={handleClick}>
        Click me
      </Button>
    )
    const button = screen.getByRole('button')
    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('applies custom styles', () => {
    render(
      <Button variant="dark" styles="custom-class">
        Styled Button
      </Button>
    )
    const button = screen.getByRole('button')
    expect(button).toHaveClass('custom-class')
  })

  it('handles disabled state', () => {
    render(
      <Button variant="dark" disabled>
        Disabled Button
      </Button>
    )
    const button = screen.getByRole('button')
    expect(button).toBeDisabled()
    expect(button).toHaveClass('bg-[#DBDEE5]')
    expect(button).toHaveClass('cursor-not-allowed')
  })

  it('maintains base styles when disabled', () => {
    render(
      <Button variant="dark" disabled>
        Disabled Button
      </Button>
    )
    const button = screen.getByRole('button')
    expect(button).toHaveClass('rounded-full')
    expect(button).toHaveClass('h-[52px]')
    expect(button).toHaveClass('w-[168px]')
  })

  it('renders children correctly', () => {
    render(
      <Button variant="dark">
        <span>Complex</span> <strong>Content</strong>
      </Button>
    )
    const button = screen.getByRole('button')
    expect(button).toHaveTextContent('Complex Content')
  })
})
