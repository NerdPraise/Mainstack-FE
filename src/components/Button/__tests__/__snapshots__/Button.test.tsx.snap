// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders with default props 1`] = `[Function]`;
